tests/cases/conformance/salsa/typeFromPropertyAssignment22.ts

```diff
@@ -0,0 +1,14 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+// @Filename: npm-install.js
+function Installer (where, dryrun, args, opts) {
+  this.args = args
+}
+Installer.prototype.loadArgMetadata = function (next) {
+  // ArrowFunction isn't treated as a this-container
+  (args) => {
+    this.args = args
+      this.newProperty = 1
```

Comment:
Include a test that `new Installer().newProperty` is `number | undefined`?
Also, fix indents

Reply:
Hm. Turns out `newProperty: number | undefined` but `args: number` even when it's initialized with `this.args = 0`. I'll have to find out why.

Reply:
Ah. The assignment in `loadArgMetadata` acts as a second declaration, so we union the types together. This is not ideal &mdash; we shouldn't redeclare properties that *know* are in the constructor &mdash; but I'll fix it separately.

Reply:
Opened #22792 to track this; there are couple of things to consider before changing this.